src/main/java/practice/StreamPractice.java

```diff
@@ -50,32 +90,27 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      */
     public List<People> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<People> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(people -> people.getAge() >= fromAge
+                        && (people.getSex().equals(People.Sex.WOMEN)
+                        ? people.getAge() <= femaleToAge
+                        : people.getAge() <= maleToAge))
+                .collect(Collectors.toList());
     }
-
+    
     /**
      * Given a List of `People` instances (having `name`, `age`, `sex` and `List of cats` fields,
      * and each `Cat` having a `name` and `age`),
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<People> peopleList, int femaleAge) {
-        return Collections.emptyList();
-    }
-
-    /**
-     * Your help with a election is needed. Given list of candidates, where each element
-     * has Candidate.class type.
-     * Check which candidates are eligible to apply for president position and return their
-     * names sorted alphabetically.
-     * The requirements are: person should be older than 35 y, should be allowed to vote,
-     * have nationality - 'Ukrainian'
-     * and live in urk for 10 years. For the last requirement use field periodsInUkr,
-     * which has following view:
-     * "2002-2015" For now we don't care if that was last 10 or not. We want to reuse our
-     * validation in future, so
-     * let's write our own impl of Predicate parametrized with Candidate in CandidateValidator.
-     */
-    public static List<String> validateCandidates(List<Candidate> candidates) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex().equals(People.Sex.WOMEN)
+                        && p.getAge() >= femaleAge
+                        && !p.getCats().isEmpty())
```

Comment:
```suggestion
        return peopleList.stream()
                .filter(p -> p.getSex().equals(People.Sex.WOMEN)
                        && p.getAge() >= femaleAge)
```
+ try to use only single map and single flatMap operation in this task

Reply:
Nice, thx)